test/integration/agent_hosting_hook_deploy_test.go

```diff
@@ -322,6 +325,48 @@ var _ = ginkgo.Describe("Agent hook deploy", func() {
 		_, err = hubWorkClient.WorkV1().ManifestWorks(hostingClusterName).
 			UpdateStatus(context.Background(), work, metav1.UpdateOptions{})
 		gomega.Expect(err).ToNot(gomega.HaveOccurred())
+
+		// addon only has pending finalizer, and status is updated
+		gomega.Eventually(func() error {
+			addon, err := hubAddonClient.AddonV1alpha1().ManagedClusterAddOns(managedClusterName).
+				Get(context.Background(), testHostedAddonImpl.name, metav1.GetOptions{})
+			if err != nil {
+				return err
+			}
+			if len(addon.Finalizers) != 1 {
+				return fmt.Errorf("addon is expected to only 1 finalizer,but got %v", len(addon.Finalizers))
+			}
+			if addon.Finalizers[0] != "pending" {
+				return fmt.Errorf("addon is expected to only pending finalizer,but got %v", len(addon.Finalizers))
+			}
+			if !meta.IsStatusConditionTrue(addon.Status.Conditions, addonapiv1alpha1.ManagedClusterAddOnHookManifestCompleted) {
+				return fmt.Errorf("addon HookManifestCompleted condition is expecte to true, but got false")
+			}
+			return nil
+		}, eventuallyTimeout, eventuallyInterval).ShouldNot(gomega.HaveOccurred())
+
```

Comment:
should we wait for 5 seconds and check if work is not recreated?

Reply:
yes, updated the addon cr label to trigger reconcile 3 times to check if the per-delete manifestwork is not re-created.